pkg/quarterdeck/db/models/apikeys.go

```diff
@@ -77,7 +77,7 @@ const (
 // empty (nil) slice if there are no results. If there is a next page of results, e.g.
 // there is another row after the page returned, then a cursor will be returned to
 // compute the next page token with.
-func ListAPIKeys(ctx context.Context, orgID, projectID ulid.ULID, prevPage *pagination.Cursor) (keys []*APIKey, cursor *pagination.Cursor, err error) {
+func ListAPIKeys(ctx context.Context, orgID, projectID, userID ulid.ULID, prevPage *pagination.Cursor) (keys []*APIKey, cursor *pagination.Cursor, err error) {
```

Comment:
I'm generally not a fan of including every single filter in the params, because it's kind of a never ending thing. I'd like to create a story for us to create a Quarterdeck query builder so that we have an easier time adding constraints (and we can do something similar in trtl for Tenant too). 

E.g. the query builder would be something like:

```
type APIKeysQuery struct {
    params []interface{}
    filters []string
    ordering string
    limit int
}

func NewAPIKeysQuery(orgID, projectID, ulid.ULID) *APIKeysQuery {

}

func (q *APIKeysQuery) WithUser(uid ulid.ULID) *APIKeysQuery {
    // and params filters
}

func (q *APIKeysQuery) WithCursor(cursor *pagination.Cursor) {
   // add cursor constraint
}
```

And then we could just do chaining:

```
NewAPIKeysQuery(orgID, projectID).WithUser(uid).WithCursor(cursor).SQL() 
```

Would you mind creating this story then pinging me and I can help fill in some of the details? The key is to figure out what queries we need builders for. Of course you could also do something more generic like:

```
type QueryBuilder struct {
    table string
    projection []string
    filters []string
    params []string
}
```

And there are tools like this; but I think something specific to our queries is ok for now.

Reply:
Yeah I would use something like that, it does feel a bit weird whenever you have to pass in multiple values of the same type into a Go method, because you don't benefit from the type checking.

Reply:
Has this story been created? If not I can create it or happy to edit it if you'd like.